mypy/types.py

```diff
@@ -2611,6 +2623,11 @@ def is_optional(t: Type) -> bool:
                                             for e in t.items)
 
 
+def is_proper_noreturn_type(t: Type) -> bool:
+    return isinstance(t, ProperType) and \
+           isinstance(t, UninhabitedType) and t.is_noreturn
```

Comment:
`t` is always `ProperType` if it is `UninhabitedType`.

I guess what we should do here is:

```
t = get_proper_type(t)
return isinstance(t, UninhabitedType) and t.is_noreturn
```

Reply:
This is related to the problem of the "Limitation" section.

```
t = get_proper_type(t)
return isinstance(t, UninhabitedType) and t.is_noreturn
```

`get_proper_type(t)` fails in  `_expand_once()` on the following assertion `assert self.alias is not None`

Therefore, I limited the implementation to not handle type aliases and added the `isinstance(t, ProperType)` check instead.